### PR TITLE
Reboot Class Diagram and rename client libraries

### PR DESCRIPTION
**Pull request by stevenhartley**
Given the final decisions for L1 and L2 APIs defined in 1.5.8 of the specification, this PR cleans up the class diagram and language vs transport libraries requirements.

### CODE REVIEW

File: languages.adoc [hunk 1]
```diff
@@ -18,88 +18,102 @@ SPDX-FileType: DOCUMENTATION
...
-| Factory methods to build the data model per business logic (i.e. to build CloudEvents, UUIDs, etc..)
+[oft-sid="req~up-language-legal~1"]
+--
+* *MUST* contain `CONTRIBUTING.adoc`, `CODE_OF_CONDUCT.adoc`, `SECURITY.adoc` that point to the versions of these files in this project.
+--
 
-|===
+[oft-sid="req~up-language-legal~2"]
+--
+* Dependencies *MUST* only be permissive licensed code/libraries (i.e. *MUST NOT* depend on external code that will pollute the license of the library).
```
[the remark this answers is not in the document]
Comment (stevenhartley) [reply languages.adoc:62]: Please share where

File: languages.adoc [hunk 2]
```diff
@@ -18,88 +18,102 @@ SPDX-FileType: DOCUMENTATION
...
- * *SHOULD* include a test suite under the folder `test`
- * *MUST* be able to run the test suite such that merge requests (commits) cannot be merged till tests are passed
-  * *SHOULD* cover 100% of all code and specifications defined in <<Source>>
 
+=== Build
 
-== Build
+[oft-sid="req~up-language-build~1"]
+--
+ * *MUST* be easily built in both a Windows and Linux development environment.
```
[the remark this answers is not in the document]
Comment (stevenhartley) [reply languages.adoc:102]: I'll remove this, it is incommensurable

File: languages.adoc [hunk 3]
```diff
@@ -18,88 +18,102 @@ SPDX-FileType: DOCUMENTATION
...
-* *MUST* fetch external dependency libraries as handle build dependencies, *MUST NOT* copy external dependencies
-* *MUST NOT* depend on external copyright or copyleft libraries/modules, only permissive licenses such as Apache 2.0, MIT, shall be accepted etc..
+[oft-sid="req~up-language-build~2"]
+--
+ * *MUST* include self contained build system to build the code and dependencies(ex. maven, gradle, clang, soong, etc...).
+--
 
+[oft-sid="req~up-language-build~3"]
+--
+ * *MUST* fetch and build external dependency libraries, *MUST NOT* copy external dependencies
```
[the remark this answers is not in the document]
Comment (stevenhartley) [reply languages.adoc:112]: Trying to say that I don't want people to copy paste dependencies into this project/library (I've seen it too many times) but instead they pull the external dependencies. I'm re-writing it now.

File: languages.adoc [hunk 4]
```diff
@@ -18,88 +18,102 @@ SPDX-FileType: DOCUMENTATION
...
+[oft-sid="req~up-language-naming~1"]
+--
+* *MUST* be named according to pattern `up-[LANGUAGE]`
+--
 
-| `serializer`
-| Any code that is responsible to Serialize and deserialize the data model
+[oft-sid="req~up-language-license~1"]
+--
+* *MUST* use a permissive (OSS) license like Apache 2.0 or MIT and located in https://github.com/eclipse-uprotocol[Eclipse uProtocol GitHub Organization] that is documented in `LICENSE` file.
```
[the remark this answers is not in the document]
Comment (sophokles73) [reply languages.adoc:52]: The [Project Handbook](https://www.eclipse.org/projects/handbook/#ip-license-compatibility) provides some guidance and refers to the [Approved Licenses List](https://www.eclipse.org/legal/licenses.php#approved).

File: languages.adoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (sophokles73) [reply languages.adoc:62]: https://www.eclipse.org/projects/handbook/#ip-license-compatibility

File: languages.adoc [hunk 5]
```diff
@@ -18,88 +18,102 @@ SPDX-FileType: DOCUMENTATION
...
+--
 
-== Documentation
+== Class Diagram
 
- * *MUST* include a main README.adoc file that explains how to build and use the library. 
- * Each module *MUST* also have a README.adoc that explains how to use said module
+.Class Diagram
+[#up-libraries]
+image:up_libraries.drawio.svg[width=800]
```
Comment (frankghe) [inline languages.adoc:119]: in the diagram, all L2 (blue) boxes are defined as interfaces, and have no implementation

File: languages.adoc [hunk 6]
```diff
@@ -18,88 +18,102 @@ SPDX-FileType: DOCUMENTATION
...
+--
 
-.Module Folders/Namespace
-[#folders,width="80%",cols="20%,80%",options="header"]
-|===
-| Folder | Description
+[oft-sid="req~up-language-comm-api~1"]
+--
+* *MUST* provide an implementation of the link:up-l2/README.adoc[uProtocol Communication Layer API] based on the uProtocol Transport Layer API.
+--
```
Comment (frankghe) [inline languages.adoc:43]: should it not also include an implementation of the L2 interfaces (that relies on transport API)

**Comment by stevenhartley** [comment]
Unfortunately have to blow away and restart my PR, I screwed something up

**Pull request closed without merging**